### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by wusticality**
This PR fixes a few things given the latest round of unit tests, namely:

* Omits certain `json` tags if unused.
* Emits user bindings in a deterministic way.
* Fixes a few bugs.

This fixes STY-13695.

### CODE REVIEW

File: api/v1/proxy/defaults.go [hunk 1]
```diff
@@ -35,15 +35,13 @@ func DefaultCallbacks(settings *DefaultCallbackSettings) *Callbacks {
 
 func NewOnModifyBatchQueryInput() OnModifyBatchQueryInput {
 	return func(session *api.Session, input interface{}) interface{} {
-		if input == nil {
-			input = make(map[string]interface{})
+		if input != nil {
```
Comment (wusticality) [inline api/v1/proxy/defaults.go:38]: This was broken before as we weren't modifying the original value.

File: api/v1/proxy/proxy.go [hunk 2]
```diff
@@ -86,8 +86,8 @@ func (p *proxy) BatchQuery() *Route {
 
 			request.Input = p.settings.Callbacks.OnModifyBatchQueryInput(session, request.Input)
 
-			for _, query := range queries {
-				query.Input = p.settings.Callbacks.OnModifyBatchQueryInput(session, query.Input)
+			for i := range queries {
```
Comment (wusticality) [inline api/v1/proxy/proxy.go:89]: This was also broken as the queries were being iterated over by value.

File: rbac/v1/rbac.go [hunk 3]
```diff
@@ -86,6 +87,10 @@ func (r *rbac) ListUserBindingsAll(ctx context.Context, session *api.Session) ([
 		)
 	}
 
+	sort.SliceStable(result, func(i, j int) bool {
```
Comment (wusticality) [inline rbac/v1/rbac.go:90]: We do this to emit user bindings in a deterministic way.

File: tests/v1/main.go [hunk 4]
```diff
@@ -7,28 +7,25 @@ import (
...
-		{Id: "kevin"},
-		{Id: "lynn"},
-		{Id: "jiri"},
-		{Id: "larry"},
-		{Id: "alan"},
-		{Id: "terence"},
-		{Id: "eckhart"},
-	}
-)
+// var (
```
Comment (mmacxm) [inline tests/v1/main.go:13]: delete instead of comment?

File: tests/v1/main.go [hunk 5]
```diff
@@ -58,13 +55,16 @@ func main() {
...
-				&rproxy.ArrayCallbackSettings{
-					GetSession: api.SessionFromCookie(),
-					Users:      users,
-					PageSize:   3,
-				},
-			),
+			RbacCallbacks: &rproxy.Callbacks{
+				GetSession: api.SessionFromCookie(),
+			},
+			// RbacCallbacks: rproxy.ArrayCallbacks(
```
Comment (mmacxm) [inline tests/v1/main.go:61]: delete this too?

**Review by mmacxm — APPROVED** [review]

File: tests/v1/main.go (hunk 5 above)
Comment (wusticality) [reply tests/v1/main.go:61]: Leaving this in until I have a solution for the user-provided user callback stuff for my own sanity :)

**Pull request merged**